### PR TITLE
Apikit 2770: Add api-autodiscovery and external common file.

### PR DESCRIPTION
**Pull request by nicolaskosacoff**

### CODE REVIEW

File: src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java [hunk 1]
```diff
@@ -110,6 +104,41 @@ private GenerationModel mockGenerationModel(String example) {
...
+  @Test
+  public void blankDocumentWithoutExternalizedGlobalsAndAPIAutodiscovery() {
+    String apiAutodiscovery = "1234";
+    ScaffoldingConfiguration.Builder builder =
+        ScaffoldingConfiguration.builder().withShowConsole(HIDE_CONSOLE).withApiAutodiscoveryId(apiAutodiscovery);
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    for (MuleConfig muleConfig : muleConfigs) {
+      Document document = muleConfig.getContentAsDocument();
+      Element rootElement = document.getRootElement();
+      assertEquals("mule", rootElement.getName());
```
Comment (JuanAller) [inline src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java:116]: where are the assertions for autodiscovery?

File: src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java [hunk 2]
```diff
@@ -110,6 +104,41 @@ private GenerationModel mockGenerationModel(String example) {
...
+    for (MuleConfig muleConfig : muleConfigs) {
+      Document document = muleConfig.getContentAsDocument();
+      Element rootElement = document.getRootElement();
+      if (muleConfig.getName() == externalConfigurationFile) {
+        assertEquals("mule", rootElement.getName());
+        assertConfigurations(rootElement);
+      } else {
+        assertMainFlow(rootElement, 0);
+      }
+      System.out.println(document);
```
Comment (JuanAller) [inline src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java:138]: ???

File: src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java [hunk 3]
```diff
@@ -110,6 +104,41 @@ private GenerationModel mockGenerationModel(String example) {
     return flowEntry;
   }
 
+  @Test
+  public void blankDocumentWithoutExternalizedGlobalsAndAPIAutodiscovery() {
```
Comment (JuanAller) [inline src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java:108]: rescaffolding a mule-config with api-autodiscovery in global and another with api-autodiscovery in main xml 
rescaffolding with mule-config with a global containing http-listener-config and apikit-config
rescaffolding with http-listener-config in global but apikit-config in main xml and inverse case
please @juanchib and @dcourtil add more test cases

File: src/main/java/org/mule/tools/apikit/model/APIAutodiscoveryConfig.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  private Boolean ignoreBasePath;
+  private String flowRef;
+
+  public APIAutodiscoveryConfig(String apiId, Boolean ignoreBasePath, String flowRef) {
+    this.apiId = apiId;
+    this.ignoreBasePath = ignoreBasePath;
+    this.flowRef = flowRef;
+  }
+
+  public APIAutodiscoveryConfig() {
```
Comment (dcourtil) [inline src/main/java/org/mule/tools/apikit/model/APIAutodiscoveryConfig.java:40]: Not needed, right? Is fo, please initialize the properties.

**Pull request closed without merging**